.github/workflows/iroha2-dev-pr-title.yml

```
@@ -3,107 +3,40 @@ name: I2::Dev::Title
-          labels: |
-            Refactor
-
-      - name: Fix
-        uses: actions-ecosystem/action-regex-match@v2
-        id: fix-match
-        if: steps.refactor-match.outputs.match == ''
-        with:
-          text: ${{ github.event.pull_request.title }}
-          regex: '^\[fix\] #\d+(, #\d+)*: .+$'
-      - name: Add fix label
-        uses: actions-ecosystem/action-add-labels@v1
-        if: steps.fix-match.outputs.match != ''
-        continue-on-error: true
-        with:
-          github_token: ${{ secrets.github_token }}
-          labels: |
-            Bug
-
-      - name: Documentation
-        uses: actions-ecosystem/action-regex-match@v2
-        id: docs-match
-        if: steps.fix-match.outputs.match == ''
-        with:
-          text: ${{ github.event.pull_request.title }}
-          regex: '^\[documentation\]( #\d+(, #\d+)*)?: .+$'
-      - name: Add Documentation label
-        uses: actions-ecosystem/action-add-labels@v1
-        if: steps.docs-match.outputs.match != ''
-        continue-on-error: true
-        with:
-          github_token: ${{ secrets.github_token }}
-          labels: |
-            Documentation
-
-      - name: CI
-        uses: actions-ecosystem/action-regex-match@v2
-        id: ci-match
-        if: steps.docs-match.outputs.match == ''
-        with:
-          text: ${{ github.event.pull_request.title }}
-          regex: '^\[ci\]( #\d+(, #\d+)*)?: .+$'
-      - name: Add CI label
-        uses: actions-ecosystem/action-add-labels@v1
-        if: steps.ci-match.outputs.match != ''
-        continue-on-error: true
-        with:
-          github_token: ${{ secrets.github_token }}
-          labels: |
-            CI
-
-      - name: Chore
-        uses: actions-ecosystem/action-regex-match@v2
-        id: chore-match
-        if: steps.ci-match.outputs.match == ''
-        with:
-          text: ${{ github.event.pull_request.title }}
-          regex: '^\[chore\]( #\d+(, #\d+)*)?: .+$'
-      - name: Add Chore label
-        uses: actions-ecosystem/action-add-labels@v1
-        if: steps.chore-match.outputs.match != ''
-        continue-on-error: true
-        with:
-          github_token: ${{ secrets.github_token }}
-          labels: |
-            chore
-
-      - name: None of the above
-        if: steps.fix-match.outputs.match == '' && steps.refactor-match.outputs.match == '' && steps.feature-match.outputs.match == '' && steps.docs-match.outputs.match == '' && steps.ci-match.outputs.match == '' && steps.chore-match.outputs.match == ''
-        run: exit 1
+      - uses: amannn/action-semantic-pull-request@v5
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - uses: marocchino/sticky-pull-request-comment@v2
+        # When the previous step fails, the workflow would stop. By adding this
+        # condition you can continue the execution with the populated error message.
+        if: always() && (steps.lint_pr_title.outputs.error_message != null)
```

```suggestion
        if: steps.lint_pr_title.outputs.error_message != null
```

isn't this equivalent?

---

I believe `always()` is needed because otherwise the check will be skipped due to the previous step failing. See more [here](https://docs.github.com/en/actions/learn-github-actions/expressions#always).